lib/src/profile/models/profile.dart

```diff
@@ -0,0 +1,112 @@
+/**
+"address": {
+"geolocation": {
+"lat": "-37.3159",
+"long": "81.1496"
+},
+"city": "kilcoole",
+"street": "new road",
+"number": 7682,
+"zipcode": "12926-3874"
+},
+"id": 1,
+"email": "john@gmail.com",
+"username": "johnd",
+"password": "m38rmF$",
+"name": {
+"firstname": "john",
+"lastname": "doe"
+},
+"phone": "1-570-236-7033",
+"__v": 0
+ */
+
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'profile.freezed.dart';
+part 'profile.g.dart';
+
+@freezed
+abstract class Profile with _$Profile {
+  const factory Profile(
+    {
+      @required Address address,
+      @required int id,
+      @required String email,
+      @required String username,
+      @required Name name,
+      @required String phone,
+      }) = _Profile;
+
+  factory Profile.fromJson(Map<String, dynamic> json) => _$ProfileFromJson(json);
+}
+
+
+class Address {
+	Geolocation geolocation;
+	String city;
+	String street;
+	int number;
+	String zipcode;
+
+	Address({this.geolocation, this.city, this.street, this.number, this.zipcode});
+
+	Address.fromJson(Map<String, dynamic> json) {
```

Comment:
Why didn't we generate the toJson and fromJson named constructors as we have done for most of our models? Is there any reason why you choose the manual approach?